### PR TITLE
add matchit 1.0.1

### PR DESCRIPTION
**Pull request by sanblch**
**matchit/1.0.1**

It's a brand new C++ header-only library that was released recently. I've seen that it only [has](https://github.com/BowenFu/matchit.cpp#option-4-manage-with-vcpkg) vcpkg support. So I decided to add conan support too as I would prefer to use it myself. It's main purpose to represent Rust-like enum pattern matchers into C++ and provide new coding style paradigm into the language.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/matchit/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    homepage = "https://github.com/BowenFu/matchit.cpp"
+    license = "Apache-2.0"
+    description = ("match(it): A lightweight header-only pattern-matching"
+                   " library for C++17 with macro-free APIs.")
+    topics = (
+        "conan",
+        "matchit",
+        "cpp17",
+        "header-only",
+    )
```
Comment (uilianries) [inline recipes/matchit/all/conanfile.py:24]:
```suggestion
    topics = ("lightweight", "cpp17", "header-only", "pattern-matching")
```
Conan and the project name are not good topics. They are focused for search engine only.

File: recipes/matchit/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    no_copy_source = True
+    settings = "compiler"
```
Comment (uilianries) [inline recipes/matchit/all/conanfile.py:26]:
```suggestion
    settings = "arch", "build_type", "compiler", "os"
```
Keep full settings always, only manage the Package ID.

File: recipes/matchit/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        copy(self, "matchit.h",
+             src=os.path.join(self.source_folder, "include"),
+             dst=os.path.join(self.package_folder, "include"))
+
+    def package_id(self):
+        self.info.clear()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "matchit"
+        self.cpp_info.names["cmake_find_package_multi"] = "matchit"
```
Comment (uilianries) [inline recipes/matchit/all/conanfile.py:71]:
```suggestion
        # TODO: Remove after Conan 2.0
        self.cpp_info.names["cmake_find_package"] = "matchit"
        self.cpp_info.names["cmake_find_package_multi"] = "matchit"

        self.cpp_info.set_property("cmake_target_name", "matchit::matchit")
        self.cpp_info.set_property("cmake_file_name", "matchit")
```
cpp_info.names should be removed in the future. set_property is the new usage.

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. 

Please, add test_v1_package based on https://github.com/conan-io/conan-center-index/tree/master/docs/package/all/test_v1_package So we can validate both generators for Conan 2.x and 1.x

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**